src/main.cpp

```diff
@@ -403,39 +403,280 @@ bool FullSyncWithDPORNodes()
+        // balance left after reserve, staking should be disabled; however, just to be safe...
+        if (BalanceAvailForStaking <= 0)
+        {
+            if (fDebug10) LogPrintf("ETTS debug: No balance available: ETTS = %f", result);
+            return result;
+        }
+
+        //reminder... void AvailableCoins(std::vector<COutput>& vCoins, bool fOnlyConfirmed=true, const CCoinControl *coinControl=NULL, bool fIncludeStakingCoins=false) const;
+        pwalletMain->AvailableCoins(vCoins, true, NULL, true);
+    }
+
+    
+    // An efficient local structure to store the UTXO's with the bare minimum info we need.
+    typedef vector< std::pair<int64_t, int64_t> > vCoinsExt;
+    vCoinsExt vUTXO;
+    // A local ordered set to store the unique "bins" corresponding to the UTXO transaction times. We are going to use this
+    // for the outer loop.
+    std::set<int64_t> UniqueUTXOTimes;
+        // We want the first "event" to be the CurrentTime. This does not have to be quantized.
+    UniqueUTXOTimes.insert(nCurrentTime);
+
+    
+    // Debug output cooldown...
+    if (fDebug10) LogPrintf("ETTS debug: nStakeMinAge = %i", nStakeMinAge);
+
+    // Derive a smoothed difficulty over desired PoSInterval of blocks by calling GetPoSKernelPS() which is netweight and reverse engineering...
+    double dDiff = GetAverageDifficulty(nPoSInterval);
+    if (fDebug10) LogPrintf("ETTS debug: dDiff = %f", dDiff);
+    
+    // The stake probability per "throw" of 1 weight unit = target value at diff of 1.0 / (maxhash * diff). This happens effectively every STAKE_TIMESTAMP_MASK+1 sec.
+    double dUnitStakeProbability = 1 / (4295032833.0 * dDiff);
+    if (fDebug10) LogPrintf("ETTS debug: dUnitStakeProbability = %e", dUnitStakeProbability);
+
+    
+    int64_t nTime = 0;
+    for (const auto& out : vCoins)
+    {
+        CTxIndex txindex;
+        CBlock CoinBlock; //Block which contains CoinTx
+        {
+            LOCK2(cs_main, pwalletMain->cs_wallet);
+            
+            if (!txdb.ReadTxIndex(out.tx->GetHash(), txindex))
+                continue; //error?
+ 
+            if (!CoinBlock.ReadFromDisk(txindex.pos.nFile, txindex.pos.nBlockPos, false))
+                continue;
+        }
+        
+        // We are going to store as an event the time that the UTXO matures (is available for staking again.)
+        nTime = (CoinBlock.GetBlockTime() & ~ETTS_TIMESTAMP_MASK) + nStakeMinAge;
+        
+        nValue = out.tx->vout[out.i].nValue;
+        
+        // Only consider UTXO's that are actually stakeable - which means that each one must be less than the available balance
+        // subtracting the reserve. Each UTXO also has to be greater than 1/80 GRC to result in a weight greater than zero in the CreateCoinStake loop,
+        // so eliminate UTXO's with less than 0.0125 GRC balances right here.
+        if(BalanceAvailForStaking >= nValue && nValue / 1250000 > 0)
```

Comment:
`&& nValue >= 1250000`?

Reply:
More efficient but a little more cryptic. I’ll change it.